cpp/src/arrow/compute/light_array.cc

```diff
@@ -480,7 +480,7 @@ void ExecBatchBuilder::Visit(const std::shared_ptr<ArrayData>& column, int num_r
       const uint8_t* field_ptr =
           column->buffers[1]->data() +
           (column->offset + row_id) * static_cast<int64_t>(metadata.fixed_length);
-      process_value_fn(i, field_ptr, metadata.fixed_length);
+      process_value_fn(i, field_ptr, static_cast<int32_t>(metadata.fixed_length));
```

Comment:
I wanted to change the type of `metadata.fixed_length` to `int32_t`, but that would bring big amount related changes overwhelming to this small PR. So I tend to leave it as is and do a simple cast here. Changing `fixed_length` of `RowTableMetadata` and `KeyColumnMetaData` to signed type could be a future enhancement.

Reply:
> Changing `fixed_length` of `RowTableMetadata` and `KeyColumnMetaData` to signed type could be a future enhancement.

Cool, thank you!